lib/modules/archivist/index.js

```diff
@@ -1,2 +1,48 @@
 // This module is a gateway between frontend's and the archivist in the
 // MAGE backend.
+
+const mage = require('lib/mage');
+const ArchivistModuleError = require('./ArchivistModuleError');
+
+exports.logger = mage.core.logger.context('archivist');
+exports.getClientVault = function (state, topic, index) {
+	let details = topic;
+
+	if (index) {
+		details = {
+			topic: topic,
+			index: index
+		};
+	}
+
+	const clientVault = state.archivist.getPrivateVault('client');
+	if (!clientVault) {
+		throw new ArchivistModuleError('Client vault not configured', state, details);
+	}
+
+	return clientVault;
+};
+
+exports.getTopicApi = function (state, clientVault, topic, index) {
+	const topicApi = state.archivist.getTopicApi(topic, clientVault.name);
+	if (!topicApi) {
+		throw new ArchivistModuleError('Topic does not exist or has no client configuration', state, {
+			topic: topic,
+			index: index
+		});
+	}
+
+	return topicApi;
+};
```

Comment:
For getTopicApi and getClientVault.. why not just have the existing functions throw? I can imagine you're trying to stay BC, but it seems a bit overkill and inconsistent to have these in 2 places.

Reply:
The goal was to make the errors being logged more consistent (e.g. if you hit archivist client API you always get more or less the same kind of errors). So it's a bit redundant, but it does currently serve a purpose.